### PR TITLE
Move trulens-core app types to trulens.apps

### PR DESCRIPTION
**Pull request by sfc-gh-chu**
# Description

keep virtual/basic/custom apps apart of trulens-core but are moved to the `trulens.instrument` namespace (previously `trulens.core.app`) to be consistent with 3rd-party app types

## Other details good to know for developers

Please include any other details of this change useful for _TruLens_ developers.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by sfc-gh-chu** [comment]
files to review
https://github.com/truera/trulens/pull/1395/files/2ea9e6d9340716c3d97eb031fb0aa8d5e1c7e3e3

File: pyproject.toml [hunk 1]
```diff
@@ -28,11 +28,12 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = "^3.9,!=3.9.7"
-trulens-core = { version = "^1.0.0", extras = [
-  "snowflake",
-], allow-prereleases = true }
+trulens-core = { version = "^1.0.0", allow-prereleases = true }
 trulens-feedback = { version = "^1.0.0", allow-prereleases = true }
 trulens-dashboard = { version = "^1.0.0", allow-prereleases = true }
+trulens-instrument-basic = { version = "^1.0.0", allow-prereleases = true }
```
Comment (sfc-gh-pmardziel) [inline pyproject.toml:34]: Unsure if they need to be seperate packages. Is the problem with including them in trulens-core is that you'd like trulens-core to only contain the folder trulens/core and not trulens/instrument/... ?

Comment (sfc-gh-pmardziel) [reply pyproject.toml:34]: And if so, can we have "core instruments" in one package `trulens-instrument-core` and include all 3 in there?

File: docs/trulens/evaluation/running_feedback_functions/existing_data.md [hunk 2]
```diff
@@ -58,7 +58,7 @@ The parameters you'll use with `VirtualRecord` are the same as those for `Record
 In the example below, we add two records. Each record includes the inputs and outputs for a context retrieval component. Remember, you only need to provide the information that you want to track or evaluate. The selectors are references to methods that can be selected for feedback, as we'll demonstrate below.
 
 ```python
-from trulens.instrument.virtual import VirtualRecord
+from trulens.instrument.core import VirtualRecord
```
Comment (sfc-gh-pmardziel) [inline docs/trulens/evaluation/running_feedback_functions/existing_data.md:61]: Bad replace?

Comment (sfc-gh-pmardziel) [reply docs/trulens/evaluation/running_feedback_functions/existing_data.md:61]: Also a few more of these.

Comment (sfc-gh-pmardziel) [reply docs/trulens/evaluation/running_feedback_functions/existing_data.md:61]: Ohh, nm. I see. Why not keep "trulens.instrument.virtual" ?

Comment (sfc-gh-pmardziel) [reply docs/trulens/evaluation/running_feedback_functions/existing_data.md:61]: I do realize if things are put into `trulens.instrument` without core, we cannot have `__init__.py` there but if we were to follow the no aliases in inits guideline, we would be ok with this.

File: src/feedback/trulens/feedback/llm_provider.py [hunk 3]
```diff
@@ -129,7 +129,7 @@ def generate_confidence_score(
         Base method to generate a score normalized to 0 to 1, used for evaluation.
 
         Args:
-            system_prompt: A pre-formatted system prompt.
+            verb_confidence_prompt: A pre-formatted system prompt.
```
Comment (sfc-gh-chu) [inline src/feedback/trulens/feedback/llm_provider.py:132]: random docs fix not related to this pr

**Review by sfc-gh-pmardziel — APPROVED** [review]

File: tests/unit/static/golden/api.trulens_eval.3.11.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 trulens_eval:
   __class__: builtins.module
-  __version__: 0.33.0
+  __version__: 1.0.0.a
```
Comment (sfc-gh-pmardziel) [inline tests/unit/static/golden/api.trulens_eval.3.11.yaml:3]: please keep this as a dump of pre 1.0.0 trulens_eval .

**Pull request merged**